src/Features/VisualBasic/Portable/AvoidUnusedMembers/VisualBasicAvoidUnusedMembersCodeFixProvider.vb

```diff
@@ -0,0 +1,40 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports Microsoft.CodeAnalysis.CodeFixes
+Imports Microsoft.CodeAnalysis.AvoidUnusedMembers
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.AvoidUnusedMembers
+    <ExportCodeFixProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeFixProviderNames.AvoidUnusedMembers), [Shared]>
+    Friend Class VisualBasicAvoidUnusedMembersCodeFixProvider
+        Inherits AbstractAvoidUnusedMembersCodeFixProvider(Of FieldDeclarationSyntax)
+
+        Protected Overrides Function GetTopmostSyntaxNodeForSymbolDeclaration(syntaxNode As SyntaxNode, isSymbolDeclarationNode As Func(Of SyntaxNode, Boolean)) As SyntaxNode
+            Dim symbolDeclNode = MyBase.GetTopmostSyntaxNodeForSymbolDeclaration(syntaxNode, isSymbolDeclarationNode)
+
+            ' Map from StatementSyntax to BlockSyntax
+            If isSymbolDeclarationNode(symbolDeclNode.Parent) Then
+                symbolDeclNode = symbolDeclNode.Parent
+            End If
```

Comment:
we have a service for this in the IDE already.  it's called ISymbolDeclarationService.

Reply:
Thanks!

---
In reply to: [212734550](https://github.com/dotnet/roslyn/pull/29511#discussion_r212734550) [](ancestors = 212734550)